libraries/AP_Mount/AP_Mount_Tusuav.cpp

```diff
@@ -0,0 +1,1109 @@
+            }
+            break;
+
+        case ParseState::Waiting_For_Total_Length:
+            if (b >= AP_MOUNT_TUSUAV_PACKETLEN_MIN) {
+                _parsed_msg.msg.total_len = b;
+                _parsed_msg.msg.data_len = _parsed_msg.msg.total_len - AP_MOUNT_TUSUAV_PACKETLEN_MIN;
+                _parsed_msg.state = ParseState::Waiting_For_Flag;
+            } else {
+                reset_parser = true;
+            }
+            break;
+
+        case ParseState::Waiting_For_Flag:
+            // length held in bits 0 ~ 5.  length includes this length byte, frame id and final crc
+            // ignore frame counter held in bits 6~7
+            _parsed_msg.msg.flag.all = b;
+            _parsed_msg.state = ParseState::Waiting_For_Cmd;
+            break;
+
+        case ParseState::Waiting_For_Cmd:
+            _parsed_msg.msg.cmd = b;
+            // _parsed_msg.data_bytes_received = 0;
+            _parsed_msg.state = ParseState::Waiting_For_Sub_Cmd;
+            break;
+
+        case ParseState::Waiting_For_Sub_Cmd:
+            _parsed_msg.msg.sub_cmd = b;
+            _parsed_msg.state = ParseState::Waiting_For_Data;
+            break;
+
+        case ParseState::Waiting_For_Data:
+            if (_msg_buff_len >= _parsed_msg.msg.total_len - 1) {
+                _parsed_msg.state = ParseState::Waiting_For_CRC;
+            }
+            break;
+
+        case ParseState::Waiting_For_CRC: {
+            _parsed_msg.msg.crc = b;
+            const uint8_t expected_crc = (uint8_t)crc_calculate(_msg_buff, _parsed_msg.msg.total_len - 1);
+            if (expected_crc == _parsed_msg.msg.crc) {
+                // successfully received a message, do something with it
+                _parsed_msg.msg.data = &_msg_buff[5];
+                process_packet();
+            } else {
+                debug("crc expected:%x got:%x data: %x %x %x %x %x %x %x", (unsigned)expected_crc, (unsigned)_parsed_msg.msg.crc,
+                _msg_buff[0],_msg_buff[1],_msg_buff[2],_msg_buff[3],_msg_buff[4],_msg_buff[5],_msg_buff[6]);
+            }
+            reset_parser = true;
+            break;
+            }
+        }
+
+        // handle reset of parser
+        if (reset_parser) {
+            _msg_buff_len = 0;
+            _parsed_msg.state = ParseState::Waiting_For_Header;
+            reset_parser = false;
+        }
+    }
+}
+
+// process successfully decoded packets held in the _parsed_msg structure
+void AP_Mount_Tusuav::process_packet()
+{
+    switch ((GimbalCmd)_parsed_msg.msg.cmd) {
+    case GimbalCmd::MasCmd_Gimbal:{
+        switch ((GimbalSubCmd)_parsed_msg.msg.sub_cmd){
+        case GimbalSubCmd::DebugSubCmd_InnerFramePIDPara:{
+            
+            break;
+        }
+
+        case GimbalSubCmd::GimbalSubCmd_GimbalAngle:{
+            MessageGimbalAngle msg;
+
+            if (_parsed_msg.msg.data_len >= sizeof(msg)) {
+                memcpy(&msg,_parsed_msg.msg.data,sizeof(msg));
```

Comment:
Cast, don't copy.

Similarly elsewhere

Reply:
Thank you for the feedback. We have replaced `memcpy` with type casting and reviewed similar cases to ensure consistency across the code.

Reply:
Thank you for the feedback. We have replaced `memcpy` with type casting and reviewed similar cases to ensure consistency across the code.

Reply:
CastThank you for the feedback. We have replaced `memcpy` with type casting and reviewed similar cases to ensure consistency across the code.

Reply:
‘xgghj’

Reply:
CastThank you for the feedback. We have replaced 'memcpy' with type casting and reviewed similar cases to ensure consistency across the code.